### PR TITLE
Testcase: sort-project-filter-scan

### PR DESCRIPTION
**Pull request by linbo-lin**

### CODE REVIEW

File: src/task/apply_rule.rs [hunk 1]
```diff
@@ -43,12 +39,14 @@ impl ApplyRuleTask {
...
-            let group_plan = optimizer_ctx.memo_mut().copy_in_plan(Some(group.clone()), &plan);
+            let group_plan = optimizer_ctx
+                .memo_mut()
+                .copy_in_plan(Some(self.plan.borrow().group().clone()), &plan);
             if group_plan.borrow().operator().is_logical() {
                 task_runner.push_task(OptimizePlanTask::new(group_plan));
             } else {
-                let required_prop: PhysicalProperties = PhysicalProperties::new();
-                let new_task = EnforceAndCostTask::new(group_plan, Rc::new(required_prop.clone()));
+                let required_prop = optimizer_ctx.required_properties();
```
Comment (davidli2010) [inline src/task/apply_rule.rs:48]: `optimizer_ctx.required_properties()` is the top required properties for root group, here we should use the required properties for current group.

File: src/rule/mod.rs [hunk 2]
```diff
@@ -168,44 +168,49 @@ impl<'a> Binding<'a> {
...
     }
 
     fn next(&mut self) -> Option<Plan> {
         if self.pattern.children().is_empty() && self.group_plan_index[0] > 0 {
             return None;
         }
 
-        let plan = None;
-
+        // todo
```
Comment (davidli2010) [inline src/rule/mod.rs]: remove `todo`

File: tests/opt.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+fn metadata_accessor() -> MdAccessor {
+    let md_cache = md_cache();
+    let md_provider = Rc::new(TempMdProvider::new(md_cache));
+    MdAccessor::new(md_provider)
+}
+
+#[test]
+fn temp() {
```
Comment (davidli2010) [inline tests/opt.rs]: give this test case a reasonable name

File: tests/opt.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    let scan = logical_scan();
+    let filter = logical_filter(vec![scan]);
+    let project = logical_project(vec![filter]);
+    let required_properties = required_properties();
+    let md_accessor = metadata_accessor();
+
+    let physical_plan = optimizer.optimize(project, required_properties, md_accessor);
+
+    dbg!(physical_plan);
```
Comment (davidli2010) [inline tests/opt.rs]: check if the result is as expected through `assert!`

**Review by davidli2010 — CHANGES_REQUESTED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**